trady_functions.lua

```diff
@@ -661,6 +662,209 @@ function StringToBool( inString )
+		addedItem.m_ItemCount = addedItem.m_ItemCount - itemsAdded
+		hotbarGrid:AddItem( addedItem )
+	end
+end
+-- This function returns item max stack for a given itemID. It uses vanilla max stack size, and uses several non-common items notations;
+-- Those are:
+-- oneonerecord( because aparently 11record wasn't the best idea in lua scripting application )
+-- carrotonastick( because it wasn't added to items.txt yet )
+-- waitrecord( for same reason )
+-- Feel free to ignore the difference, or to add those to items.txt
+function GetItemMaxStack( inItemID )
+	local testerItem = cItem( inItemID )
+	LOGINFO( "This function serves no real purpose now, maybe consider using cItem:GetMaxStackSize()?" )
+	return testerItem:GetMaxStackSize()
+end
+function ItemIsArmor( inItemID, inCheckForHorseArmor )
+	inCheckForHorseArmor = inCheckForHorseArmor or false
+	if( inItemID == E_ITEM_LEATHER_CAP )		then	return true		end
+	if( inItemID == E_ITEM_LEATHER_TUNIC )		then	return true		end
+	if( inItemID == E_ITEM_LEATHER_PANTS )		then	return true		end
+	if( inItemID == E_ITEM_LEATHER_BOOTS )		then	return true		end
+	
+	if( inItemID == E_ITEM_CHAIN_HELMET )		then	return true		end
+	if( inItemID == E_ITEM_CHAIN_CHESTPLATE )	then	return true		end
+	if( inItemID == E_ITEM_CHAIN_LEGGINGS )		then	return true		end
+	if( inItemID == E_ITEM_CHAIN_BOOTS )		then	return true		end
+	
+	if( inItemID == E_ITEM_IRON_HELMET )		then	return true		end
+	if( inItemID == E_ITEM_IRON_CHESTPLATE )	then	return true		end
+	if( inItemID == E_ITEM_IRON_LEGGINGS )		then	return true		end
+	if( inItemID == E_ITEM_IRON_BOOTS )			then	return true		end
+	
+	if( inItemID == E_ITEM_DIAMOND_HELMET )		then	return true		end
+	if( inItemID == E_ITEM_DIAMOND_CHESTPLATE )	then	return true		end
+	if( inItemID == E_ITEM_DIAMOND_LEGGINGS )	then	return true		end
+	if( inItemID == E_ITEM_DIAMOND_BOOTS )		then	return true		end
+	
+	if( inItemID == E_ITEM_GOLD_HELMET )		then	return true		end
+	if( inItemID == E_ITEM_GOLD_CHESTPLATE )	then	return true		end
+	if( inItemID == E_ITEM_GOLD_LEGGINGS )		then	return true		end
+	if( inItemID == E_ITEM_GOLD_BOOTS )			then	return true		end
+	
+	if( inCheckForHorseArmor ) then
+		if( inItemID == E_ITEM_IRON_HORSE_ARMOR )		then	return true		end
+		if( inItemID == E_ITEM_GOLD_HORSE_ARMOR )		then	return true		end
+		if( inItemID == E_ITEM_DIAMOND_HORSE_ARMOR )	then	return true		end
+	end
+	return false
+end
+-- Returns full-length playername for a short name( usefull for parsing commands )
+function GetExactPlayername( inPlayerName )
+	local _result = inPlayerName
+	local function SetProcessingPlayername( inPlayer )
+		_result = inPlayer:GetName()
+	end
+	cRoot:Get():FindAndDoWithPlayer( inPlayerName, SetProcessingPlayername )
+	return _result
+end
+function GetPlayerByName( inPlayerName )
+	local _player
+	local PlayerSetter = function( Player )
+		_player = Player
+	end
+	cRoot:Get():FindAndDoWithPlayer( inPlayerName, PlayerSetter )
+	return _player
```

Comment:
Instead of
```lua
GetPlayerByName(name):DoSomething()
```
use
```lua
cRoot:Get():FindAndDoWithPlayer(name,
  function(a_CBPlayer)
    a_CBPlayer:DoSomething()
  end
)
```
If the usage is all the same (sending a message) all the time, consider wrapping that into a helper function instead (`SendMessageToPlayer(a_PlayerName, a_Message)`)